### PR TITLE
feat(orchestration): calibrate outcome verification

### PR DESCRIPTION
**Pull request by Nanako0129**
## Summary

- add calibrated `CONFIRMED`, `REFUTED`, and `INCONCLUSIVE` verifier outcomes
- require reproducible P0-P2 claim blockers for `REFUTED`; keep P3/P4 advisory
- move final finding disposition to the main session
- bound P1 recovery at five materially changed passes while preserving the two-pass Plan-readiness cap
- define explicit `AUTO` / `ASK` behavior for long autonomous runs without expanding authority

## Why

The previous binary verifier contract encouraged models to treat every concern as a blocker and could trap long-running work in repeated fix/reverify cycles. Large plans naturally expose more details, so verification needs impact calibration, an inconclusive state, and a bounded exit that pauses only affected slices when risk is local.

The prompt-compression provenance test now reads the immutable gate snapshot rather than current policy templates. This preserves historical benchmark evidence while allowing orchestration policy to evolve.

## Impact

Verifiers now report structured priority, confidence, evidence, expected/actual behavior, and recheck instructions. The main session decides whether to fix, defer, narrow the claim, or pause. Long-running work announces its interaction mode, and neither mode creates commit, publish, install, destructive, external-action, or spending authority.

## Verification

- 31 policy tests passed
- `git diff --check`
- paid Claude and Grok verifier/ASK smokes
- exact Grok severity regression rechecked as P2 after calibration
- fresh cross-project outcome verifier: `CONFIRMED`

Paid smoke total across the four companion implementations: **US$0.9722691**. The Codex smoke reported 30,438 input and 60 output tokens but no dollar field, so it is excluded from that total.

### CODE REVIEW

File: templates/agents/verifier.md [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
 model: opus
 effort: medium
 disallowedTools: Write, Edit, NotebookEdit, Agent, Workflow
 ---
 
 Leaf agent: do whole task yourself, this session. Never delegate — Agent/Workflow tools disabled by design. Task seems to need sub-agents → mis-routed, stop and report back.
 
-Adversarial outcome verifier with fresh eyes. Receive claim ("X was implemented and works") plus relevant diff or paths. Try to REFUTE it: independently run tests, drive affected flow, probe plausible edge cases, inspect what diff doesn't handle. Don't trust implementer's own test run; reproduce it.
+Fresh-context outcome verifier. Receive the exact claim and acceptance plus relevant diff or paths. Independently reproduce relevant checks, drive the affected flow, and inspect claim-relevant edge cases and diff coverage. Report only reproducible issues relevant to the exact claim.
```
[the remark this answers is not in the document]
Comment (Nanako0129) [reply templates/agents/verifier.md:12]: The verifier prompt and main-session disposition policy now state that regressions caused by the reviewed implementation are claim-relevant even when the brief did not name the affected flow. The policy test locks this behavior.

File: templates/claude-md.orchestration.md [hunk 2]
```diff
@@ -31,12 +31,23 @@ Delegation rules:
...
 - Model routing owned by agent definitions. Invoking named role: omit `model` arg — invocation-level model overrides role routing.
 - `model` only for truly ad-hoc agent, no named role — never let it inherit main-session model accidentally.
-- `plan-verifier` brief: bare **READY** or structured **REVISE** (`Blocker:`, `Evidence:`, `Minimum revision:`, `Acceptance check:`) for one stable envelope/slice. Malformed output = protocol failure, not Plan judgment. Outcome `verifier` brief: **CONFIRMED**/**REFUTED** only. Never swap roles — Plan role read-only tools, outcome role keeps Bash for test reproduction post-approval.
+- `plan-verifier` brief: bare **READY** or structured **REVISE** (`Blocker:`, `Evidence:`, `Minimum revision:`, `Acceptance check:`) for one stable envelope/slice. Malformed output = protocol failure, not Plan judgment. Outcome `verifier` brief: exact claim/acceptance + relevant diff/paths; **CONFIRMED**/**REFUTED**/**INCONCLUSIVE**. Never swap roles — Plan role read-only tools, outcome role keeps Bash for test reproduction post-approval.
```
[the remark this answers is not in the document]
Comment (Nanako0129) [reply templates/claude-md.orchestration.md:34]: Set VERSION and the installed policy stamp to 1.3.5, added the v1.3.5 changelog entry, and updated both pinned installation paths and their contract test. Existing v1.3.4 installs are now detected as older instead of being treated as current.

File: templates/claude-md.orchestration.md [hunk 3]
```diff
@@ -31,12 +31,23 @@ Delegation rules:
...
+- P0 freezes the affected slice and dependents; a cross-cutting P0 stops the program. Automatic containment is limited to agent-owned work/evidence, never external action.
+- P1 recovery allows at most five meaningful fix/reverify passes per stable slice/claim: rounds 1-2 normal, 3-5 recovery. Each next pass requires material implementation, claim, contract, evidence, or environment change; never reverify the same head/claim/environment. After five, mark that slice `PAUSED_VERIFICATION`, block dependents, and continue unrelated approved slices only when risk is not cross-cutting.
+- P2 joins the next coherent integration-boundary verification; P3/P4 get no dedicated loop. Stop the whole run only for a cross-cutting blocker, all remaining work depending on a paused slice, new authority/product decision, destructive/irreversible/external action, exhausted budget/quota, unsafe environment, or unattainable original scope.
```
[the remark this answers is not in the document]
Comment (Nanako0129) [reply templates/claude-md.orchestration.md]: Blocking P1 and P2 findings now share one five-pass per-slice/claim recovery budget. Each retry also requires a changed candidate-state fingerprint, claim, or environment; P3/P4 still receive no dedicated loop.

File: templates/agents/verifier.md [hunk 4]
```diff
@@ -1,19 +1,25 @@
...
+- **REFUTED** — at least one reproducible P0-P2 finding blocks the exact claim. For each finding or advisory state Priority P0-P4, Confidence high/medium/low, Evidence, Expected, Actual, and Recheck. P3/P4 are non-blocking advisories and cannot by themselves produce REFUTED.
+- **INCONCLUSIVE** — evidence, environment, or contract is insufficient or unsafe. State the reason, missing evidence, and retry condition. Lack of evidence is neither false CONFIRMED nor speculative REFUTED.
 
-Work under verification security-sensitive (authn/authz, secrets, crypto, validation) → be exhaustive rather than economical: probe abuse cases and trust-boundary bypasses, not just functional edge cases, treat as maximum-thoroughness pass.
+Priority measures real user/system impact, not claim centrality: P0 = data loss, credential/secret exposure, auth bypass, irreversible destructive action, or broad outage; P1 = reproducible high-impact security/correctness failure; P2 = material bounded/recoverable issue; P3 = minor issue; P4 = advisory/speculation. A failed acceptance condition is P2 when bounded/recoverable unless it independently meets P0 or high-impact P1 criteria.
```
[the remark this answers is not in the document]
Comment (Nanako0129) [reply templates/agents/verifier.md]: Expanded P1 to cover any reproducible high-impact user or system failure, explicitly including security, correctness, performance, reliability, and resource-cost regressions. The verifier contract and policy tests now enforce the broader taxonomy.

File: templates/claude-md.orchestration.md [hunk 5]
```diff
@@ -31,12 +31,23 @@ Delegation rules:
...
+- `ASK`: use `AskUserQuestion` when available in the current Claude Code mode. Otherwise end the turn with `PAUSED_NEEDS_USER`, one concise question, choices, and a recommended choice. Headless/noninteractive execution emits `PAUSED_NEEDS_USER` and exits; never poll, retry, guess, or continue the affected slice. Asking belongs to the main session, never a child.
+- P0 freezes the affected slice and dependents; a cross-cutting P0 stops the program. Automatic containment is limited to agent-owned work/evidence, never external action.
+- Blocking P1/P2 recovery shares at most five meaningful fix/reverify passes per stable slice/claim and acceptance: rounds 1-2 normal, 3-5 recovery. Each next pass requires material implementation, claim, acceptance, contract, evidence, or environment change. Track a candidate-state fingerprint containing the committed head plus working-tree diff or tested-artifact digest; never reverify the same fingerprint/claim/acceptance/environment. After five, mark that slice `PAUSED_VERIFICATION`, block dependents, and continue unrelated approved slices only when risk is not cross-cutting.
```
[the remark this answers is not in the document]
Comment (Nanako0129) [reply templates/claude-md.orchestration.md]: The candidate fingerprint now covers committed head, tracked and staged changes, and untracked input paths plus content; a tested-artifact digest may replace the complete input fingerprint. New untracked source or configuration inputs therefore change the verification identity.

File: templates/claude-md.orchestration.md [hunk 6]
```diff
@@ -31,12 +31,23 @@ Delegation rules:
...
+- Final disposition stays main-session-owned. For every finding assess reproducibility, whether introduced/in scope, exact claim relevance, priority, and confidence. A regression caused by the reviewed implementation is claim-relevant even when the brief did not name the affected flow. P0: freeze the affected slice and pause/ask; automatic work is containment only. P1: fix within approved scope or pause/ask. An introduced P2 regression remains blocking and must be fixed within approved scope or paused; other P2: fix when within explicit acceptance and bounded, otherwise defer with rationale and narrow the final claim if needed. Never silently defer, reject, downgrade, or call a blocker fixed without contrary evidence or a successful recheck of the original failure. P3/P4: default defer/report, never open an immediate fix-reverify loop. INCONCLUSIVE: retry once only after prerequisites materially change; otherwise pause the affected slice.
```
[the remark this answers is not in the document]
Comment (Nanako0129) [reply templates/claude-md.orchestration.md]: The main session may now document a regrade directly from the verifier cited evidence when that evidence establishes different impact. Silent downgrades remain prohibited, while a reproducible failure no longer needs to disappear before its priority can be corrected.

File: templates/agents/verifier.md [hunk 7]
```diff
@@ -1,19 +1,27 @@
...
 
-Return **CONFIRMED** or **REFUTED** only. Refutation includes exact failure scenario, expected versus actual behavior, where it breaks.
+Return one calibrated verdict:
 
-Never plan, edit, or fix anything — even one-line change. Value is independence; main-session orchestrator owns Plan synthesis and routes fixes.
+- **CONFIRMED** — evidence independently produced or inspected in this session is sufficient for every required acceptance condition. List each condition checked and its evidence/result. May include clearly non-blocking advisories.
+- **REFUTED** — at least one reproducible P0-P2 finding blocks the exact claim. P3/P4 are non-blocking advisories and cannot by themselves produce REFUTED.
+- **INCONCLUSIVE** — evidence, environment, or contract is insufficient or unsafe. State the reason, missing evidence, and retry condition. Lack of evidence is neither false CONFIRMED nor speculative REFUTED.
```
[the remark this answers is not in the document]
Comment (Nanako0129) [reply templates/agents/verifier.md:17]: Added explicit precedence: REFUTED wins when a reproducible P0-P2 blocker coexists with missing evidence, and the verifier must report both. Without a known blocker, any unevaluated required acceptance condition produces INCONCLUSIVE.

File: templates/claude-md.orchestration.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (Nanako0129) [reply templates/claude-md.orchestration.md]: The single INCONCLUSIVE retry now applies after the stated missing evidence, contract, prerequisite, or environment materially changes. The retry condition therefore matches every remediable cause represented in the verification identity.

File: templates/claude-md.orchestration.md [hunk 8]
```diff
@@ -31,12 +31,23 @@ Delegation rules:
...
+- P0 freezes the affected slice and dependents; a cross-cutting P0 stops the program. Automatic containment is limited to agent-owned work/evidence, never external action.
+- Blocking P1/P2 recovery shares at most five meaningful fix/reverify passes: rounds 1-2 normal, 3-5 recovery. Each next pass requires a material change to the stable verification identity: candidate, claim, acceptance, contract, available evidence/prerequisites, or environment. Fingerprint the complete tested candidate from committed head, tracked/staged diff, and untracked input paths plus content; a tested-artifact digest may replace that input fingerprint. Never reverify the same complete identity. After five unsuccessful or still-blocking passes, mark that slice `PAUSED_VERIFICATION`, block dependents, and continue unrelated approved slices only when risk is not cross-cutting.
```
[the remark this answers is not in the document]
Comment (Nanako0129) [reply templates/claude-md.orchestration.md]: Recovery identity now changes only with the candidate, claim, acceptance, contract, genuinely external evidence or prerequisites, or environment. The immediately preceding verifier verdict/output is explicitly excluded as new evidence, and the policy test covers that exclusion. The full 31-test suite passes.

[the remark this answers is not in the document]
Comment (Nanako0129) [reply templates/claude-md.orchestration.md]: The section now states that its recovery budget and severity rules apply to every verification run; only AUTO/ASK mode selection is limited to likely long autonomous work. All ordinary and long-run P1/P2 recovery therefore shares the same five-pass cap. The full 31-test suite passes.
